src/Couchbase.Lite.Shared/API/Error/CouchbaseLiteErrorMessage.cs

```diff
@@ -0,0 +1,84 @@
+        //Replicator Start()
+        internal const string ReplicatorDisposed = "Replication cannot be started after disposal";
+        //Replicator ResetCheckpoint()
+        internal const string ReplicatorNotStopped = "Replicator is not stopped.Resetting checkpoint is only allowed when the replicator is in the stopped state.";
+        //Query Parameters SetValue
+        internal const string QueryParamNotAllowedContainCollections = "Query parameters are not allowed to contain collections";
+        //Query Join
+        internal const string MissASforJoin = "Missing AS clause for JOIN";
+        internal const string MissONforJoin = "Missing ON statement for JOIN";
+        //Query QueryExpression
+        internal const string ExpressionsMustBeIExpressionOrString = "Expressions must either be IExpression or string";
+        internal const string InvalidExpressionValueBetween = "Invalid expression value for expression1 of Between({0})";
+        //Query QueryResultSet
+        internal const string ResultSetAlreadyEnumerated = "This result set has already been enumerated, please re-run Execute() on the original query";
+        //Query
+        internal const string ExpressionsMustContainOnePlusElement = "{0} expressions must contain at least one element";
+        internal const string DuplicateSelectResultName = "Duplicate select result named {0}";
+        internal const string NoAliasInJoin = @"The default database must have an alias in order to use a JOIN statement 
+                                                (Make sure your data source uses the As() function)";
+        internal const string InvalidQueryDBNull = "Invalid query, Database == null";
+        internal const string InvalidQueryMissingSelectOrFrom = "Invalid query, missing Select or From";
+        //MArray MDict <--- in the future, we will replace them with LiteCore Mutable Fleece API
+        internal const string CannotRemoveItemsFromNonMutableMArray = "Cannot remove items from a non-mutable MArray";
+        internal const string CannotRemoveStartingFromIndexLessThan = "Cannot remove starting from an index less than 0 (got {0})";
+        internal const string CannotRemoveRangeEndsBeforeItStarts = "Cannot remove a range that ends before it starts (got start= {0}, count = {1} )";
+        internal const string RangeEndForRemoveExceedsArrayLength = "Range end for remove exceeds the length of the array(got start = {0}, count = {1} )";
+        internal const string CannotSetItemsInNonMutableMArray = "Cannot set items in a non-mutable MArray";
+        internal const string CannotClearNonMutableMArray = "Cannot clear a non-mutable MArray";
+        internal const string CannotInsertItemsInNonMutableMArray = "Cannot insert items in a non-mutable MArray";
+        internal const string CannotClearNonMutableMDict = "Cannot clear a non-mutable MDict";
+        internal const string CannotSetItemsInNonMutableInMDict = "Cannot set items in a non-mutable MDict";
```

Comment:
@bmeike The bundle idea is good, but I feel it is overkill for just less than 50 error messages. Also in the future the number of error messages will go further down because some of them will be coming from LiteCore instead.... But let's vote for it and check the doability of it. cc. @jayahariv

Reply:
Ok.  What does "overkill" mean?  Is it too hard to maintain?  Is there some downside that I've missed?

Reply:
The xml schema for each platform will not be exactly the same, so some modification will be needed. So out of box shareable is not possible.

Reply:
For .Net, I believe I need to prepare different ways to support Resource Bundle for each platforms as well.

Reply:
Why?  I cannot see any reason that that would be the case...

Reply:
The xml schema for each platform will not be exactly the same.

Reply:
I hear you say that, but I don't understand why it is true.  Why could they not all be the same?  At the very worst, we could write little perl/Python programs that generated per-platform implementations from the common source of truth...

Reply:
The resource bundle xml (xaml for .Net) schema are not be the same. I know from my past experience.....

Reply:
Got it.  That is easy to fix:
- Don't use a standard resource bundle.  Some small simple parsing (you already have a JSON parser...) would do it.
- Like I suggested, autogenerate the per-platform resource bundles from a single source of truth.  It would take, like 20 lines of perl or python

Reply:
If we are autogenerating then why not just autogenerate the file above from our data source.  Adding in a resource bundles when we have the ability to generate per platform source simply adds complexity.  I bet I could even write a little .NET program that transforms the C# file into JSON or XML or YAML or even INI.

Reply:
RIght.  I don't care if the source of truth is one of the platform native files.  This one is annoying but not impossible to parse.  Though I can't see any reason we'd convert it to XML or YAML.  Nobody can use those.  I just want a single source of error messages.